src/app/public-comment/public-comment.component.ts

```diff
@@ -0,0 +1,48 @@
+import { ChangeDetectorRef, ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { PublicComment } from '../models/publiccomment';
+import { Subscription } from 'rxjs/Subscription';
+import { PaginationInstance } from 'ngx-pagination';
+
+@Component({
+  selector: 'app-public-comment',
+  templateUrl: './public-comment.component.html',
+  styleUrls: ['./public-comment.component.scss']
+})
+export class PublicCommentComponent implements OnInit {
+  project: PublicComment;
+  public loading: boolean;
+  public isDesc: boolean;
+  public column: string;
+  public direction: number;
+
+  public config: PaginationInstance = {
+    id: 'custom',
+    itemsPerPage: 10,
+    currentPage: 1
+  };
+
+  private sub: Subscription;
+
+  constructor(private _changeDetectionRef: ChangeDetectorRef, private route: ActivatedRoute, private router: Router) { }
+
+  ngOnInit() {
+    this.loading = true;
+    this.sub = this.route.data.subscribe(
+      (data: { project: PublicComment }) => {
+        this.project = new PublicComment(data.project);
+
+        this.column = 'dateAdded';
+        this.direction = -1;
+        // Needed in development mode - not required in prod.
+        this._changeDetectionRef.detectChanges();
```

Comment:
Does this get automatically removed in the **prod** version of the code?

Reply:
Fixed. Deleted it and did nothing so far...